tests/functional/tests.py

```diff
@@ -145,40 +145,103 @@ def json_report(self, err_msg='', alert='', status=''):
 def generate_bucket_name():
     return "minio-py-test-" + uuid.uuid4().__str__()
 
-def test_make_bucket(client, log_output):
+def is_s3(client):
+    return "s3.amazonaws" in client._endpoint_url
+
+def test_make_bucket_default_region(client, log_output):
     # Get a unique bucket_name
     log_output.args['bucket_name'] = bucket_name = generate_bucket_name()
-
-    is_s3 = client._endpoint_url.startswith("s3.amazonaws")
+    # Default location
+    log_output.args['location'] = "default value ('us-east-1')"
     try:
-        # Create a bucket
+        # Create a bucket with default bucket location
         client.make_bucket(bucket_name)
         # Check if bucket was created properly
+        log_output.function = 'bucket_exists(bucket_name)'
```

Comment:
Isn't it little confusing to change function name multiple times in the same test function ? I thought log_output.function should represent the name of the function.

Reply:
It is the name of the API being tested

Reply:
If you check the end of `tests.py` file (starting with line#1555), you notice, we first create an instance of `LogOutput` class, `log_output`, and we initialize the `function` attribute with the name of the api command/method that will go under test.

For example, in the above test case `test_make_bucket_default_region`, `log_output.function` is set to `make_bucket(bucket_name, location)` by default.

Because some tests, like `test_make_bucket_default_region`, depend on some other `helper` api commands, it would be misleading to report a helper command ,`bucket_exists`, failure with the name of the api command, `make_bucket`, under test. That's why, I  set `log_output.function` to 'bucket_exists(bucket_name)'` before calling `bucket_exists` api/helper command. The same with `remove_bucket` helper api command. This way, the name of the failed api command  will be reflected in the generated Json format output in the `function` field.

Since the code is not clear enough, (@vadmeste and I exchanged ideas and we agreed on) it would be helpful to add comments that explain the default value of `log_output.function` in each test definition.

Reply:
Done.

Reply:
we can simplify the code in another PR